packages/metablocks/src/api.ts

```diff
@@ -140,21 +145,65 @@ const withdrawNft = async (args: WithdrawNftApiArgs) => {
   try {
     const program = getMetaBlocksProgram(args.connection, args.wallet);
     const usersKey = args.wallet.publicKey;
-    const withdrawNftInstruction = await getWithdrawNftInstruction({
-      program: program,
-      usersKey: usersKey,
-      mintKey: args.mintKey,
-      universeKey: args.universeKey,
-    });
-    const transaction = new Transaction();
-    transaction.add(withdrawNftInstruction);
+    return await callWithdrawNft(
+      program,
+      usersKey,
+      args.mintKey,
+      args.universeKey
+    );
+  } catch (e) {
+    throw new KyraaError(e);
+  }
+};
+
+const withdrawNftWithReceipt = async (args: WithdrawNftWithReceiptApiArgs) => {
+  try {
+    const program = getMetaBlocksProgram(args.connection, args.wallet);
+    const usersKey = args.wallet.publicKey;
+
+    const userNftAccount: any = await accountApi.getUserNft(
+      program,
+      args.receiptMint,
+      usersKey
+    );
+
+    if (
+      userNftAccount == null ||
+      userNftAccount.tokenMint === undefined ||
+      userNftAccount.universe === undefined
+    ) {
+      throw new Error(
+        'userNftAccount failure, either token mint or universe is undefined'
```

Comment:
Ideally, these errors should have a code. And its logistically easier to have the text version defined in a JSON file.